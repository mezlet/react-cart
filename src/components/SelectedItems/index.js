import React from 'react';
import { ButtonGroup, Button } from 'react-bootstrap';

const SelectedItems = ({item, increase, decrease, removeItem, symbol, cache }) => {
    return (
        <div className='bg-white my-2 p-3'>
            <div className='d-flex justify-content-between'>
                <h6>{item.title}</h6>
                <Button className="close" onClick={removeItem} data-test='remove-item'>
                    <span aria-hidden="true">&times;</span>
                </Button>
            </div>

            <div className='my-3 px-5 d-flex justify-content-end'>
                <img src={item.image_url} alt=""  width='20' height='20' data-test='product-image' />
            </div>

            <div className='d-flex my-3'>
                <ButtonGroup size='sm' data-test='cart-button-group'>
                    <Button variant="light" onClick={decrease}>-</Button>
                    <Button variant="light" className='amount button'>{cache[item.id]}</Button>
                    <Button variant="light" onClick={increase}>+</Button>
                </ButtonGroup>
                <div className='mx-5 px-5'>{symbol}{Number(item.price) * Number(cache[item.id])}</div>
            </div>
        </div>
    )
}

export default SelectedItems
