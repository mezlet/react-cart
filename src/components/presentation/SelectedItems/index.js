import React from 'react';
import { ButtonGroup, Button } from 'react-bootstrap';

const SelectedItems = ({item, increase, decrease, removeItem, symbol }) => {
    return (
        <div className='bg-white my-2 p-3'>
            <div className='d-flex justify-content-between'>
                <h6>{item.title}</h6>
                <Button className="close" onClick={removeItem}>
                    <span aria-hidden="true">&times;</span>
                </Button>
            </div>

            <div className='my-3 px-5 d-flex justify-content-end'>
                <img src={item.image_url} alt=""  width='20' height='20' />
            </div>

            <div className='d-flex my-3'>
                <ButtonGroup size='sm' >
                    <Button variant="light" onClick={decrease}>-</Button>
                    <Button variant="light" className='amount button'>{item.count}</Button>
                    <Button variant="light" onClick={increase}>+</Button>
                </ButtonGroup>
                <div className='mx-5 px-5'>{symbol}{item.total}</div>
            </div>
        </div>
    )
}

export default SelectedItems
