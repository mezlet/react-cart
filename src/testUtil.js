import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
    return checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
}