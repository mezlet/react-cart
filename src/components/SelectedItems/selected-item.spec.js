import React from 'react';
import { shallow } from 'enzyme';
import SelectedItems from './index';
import { checkProps} from '../../testUtil';

const props = {
    item: {}, 
    increase: jest.fn(), 
    decrease: jest.fn(), 
    removeItem: jest.fn(), 
    symbol: '', 
    cache: {}
}

const setUp = (props = {}) => {
    const component = shallow(<SelectedItems {...props} />);
    return component;
}

describe('SelectedItems', () => {
    let component;
    beforeEach(() => {
        component = setUp(props);
    })
    it('should render without errors', () => {
        expect(component).toMatchSnapshot(); 
    });
    it('should render close button', () => {
        const wrapper = component.find(`[data-test='remove-item']`);
        expect(wrapper.length).toBe(1); 
    });
    it('should render product image', () => {
        const wrapper = component.find(`[data-test='product-image']`);
        expect(wrapper.length).toBe(1); 
    });
    it('should render button group', () => {
        const wrapper = component.find(`[data-test='cart-button-group']`);
        expect(wrapper.length).toBe(1); 
    });
    it('Checking PropTypes should not throw error', () => {
        const propErr = checkProps(SelectedItems, props);
        expect(propErr).toBeUndefined();
    });
    it('should remove item from cart', () => {
        const button = component.find(`[data-test='remove-item']`);
        button.simulate('click');
        expect(props.removeItem.mock.calls.length).toBe(1);
    })
})