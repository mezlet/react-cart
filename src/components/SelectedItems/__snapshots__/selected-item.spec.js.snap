// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItems should render without errors 1`] = `ShallowWrapper {}`;
